src/btouch.cs

```diff
@@ -246,7 +246,7 @@ static int Main2 (string [] args)
 			{ "core", "Use this to build product assemblies", v => Generator.BindThirdPartyLibrary = false },
 			{ "r=", "Adds a reference", v => references.Add (v) },
 			{ "lib=", "Adds the directory to the search path for the compiler", v => libs.Add (StringUtils.Quote (v)) },
-			{ "compiler=", "Sets the compiler to use", v => compiler = v },
+			{ "compiler=", "Sets the compiler to use (Obsolete) ", v => { Console.WriteLine ("The --compiler option is obsolete and ignored."); }, true },
```

Comment:
I don't think we should remove the --compiler functionality, that removes an escape hatch if someone needs to use mcs for whatever reason (it also removes the possibility for using a custom script that adds additional arguments as the compiler, and ends up calling csc anyway).

Reply:
I did that because @spouliot said if I kept it we'd have to write tests both for mcs and csc, and I didn't want to have to maintain it.

If we need to do that, I can (or we can leave it untested).

Thoughts?

Reply:
I'm fine with hiding it and saying it's obsolete (so no testing), but still keep it working so that we have an escape hatch if we ever need it.

Reply:
Good call. We'll need to use a custom `csc` in the near future to prototype, so the option will be useful. However the option should only mean the compiler path to be used (i.e. not use the default, system mono `csc`) and not that the tool itself knows about different compilers.

IOW if extra logic is needed (or differ from `csc`) it's up to the caller to write a script to handle the differences.